UI.lua

```diff
@@ -1330,8 +1331,16 @@ do
                                 end
 
                                 if i == 1 and conf.delays.fade then
-                                    local delay = b.ExactTime and ( b.ExactTime - now ) or 0
-                                    --[[ local start, duration = 0, 0
+                                    local delay = 0
+                                    if conf.delays.extend and conf.delays.onlyFadeIfExtended then
```

Comment:
Help me think through the logic here; why are we modifying the delay itself?  Also, don't need `== true` and we should be able to do this without nested `if` statements given the delay is set to the same value in either case.

Reply:
Is this stateful code where you set a flag in `RefreshCooldowns()` and check it in `OnUpdate()` actually needed? There seems to be commented-out code here that could be adapted to do this without the extra state.